### DISCUSSION TITLE
Using SpotBugs as a maven dependency

### DISCUSSION THREAD

**Discussion by IkeVoodoo** [spotbugs/spotbugs#2357] [Q&A]
Hello, I'd like to use SpotBugs as a library rather than a standalone program, I included it with

```xml
<dependencies>
    <dependency>
        <groupId>com.github.spotbugs</groupId>
        <artifactId>spotbugs</artifactId>
        <version>4.7.3</version>
    </dependency>
</dependencies>
```

However I can't figure out where I can pass in either bytecode or source code to scan, any help?

**Comment by hazendaz** [comment]
Why not just use the maven plugin?

Sent from my Verizon, Samsung Galaxy smartphone
Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: IkeVoodoo ***@***.***>
Sent: Monday, February 20, 2023 8:02:28 AM
To: spotbugs/spotbugs ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [spotbugs/spotbugs] Using SpotBugs as a maven dependency (Discussion #2357)


Hello, I'd like to use SpotBugs as a library rather than a standalone program, I included it with

<dependencies>
    <dependency>
        <groupId>com.github.spotbugs</groupId>
        <artifactId>spotbugs</artifactId>
        <version>4.7.3</version>
    </dependency>
</dependencies>

However I can't figure out where I can pass in either bytecode or source code to scan, any help?

**Comment by IkeVoodoo** [comment]
I need to run it at runtime for some dynamically loaded JARs (I do also get provided the sources because they are bundled in the jar)

**Comment by hazendaz** [comment]
Ok you would need to look at the spotbugs maven plugin or the gradle maven plugin or even look at the various existing documentation to see how you would do that from a command line, all of which will show you exactly how to do what you want.  I'm closing this otherwise as it should be relatively easy and its understood what you are after is how the standalone and plugins actually work.
